web/packages/build/eslint.config.mjs

```diff
@@ -151,10 +150,12 @@ export default tseslint.config(
       'react/no-unescaped-entities': 'warn',
       'react/jsx-key': 'warn',
       'react/jsx-no-target-blank': 'warn',
-
-      'react-hooks/rules-of-hooks': 'warn',
-      'react-hooks/exhaustive-deps': 'warn',
-      'react-hooks/react-compiler': 'warn',
+      // Enable recommended react-hooks rules as warnings.
+      ...Object.fromEntries(
+        Object.entries(reactHooksPlugin.configs.recommended.rules).map(
+          ([ruleName]) => [ruleName, 'warn']
+        )
+      ),
```

Comment:
It's not pretty but it does the job. You can check what errors it reports by changing `warn` to `error` and re-running ESLint.

Reply:
For some reason the rules are much more aggressive now. For example, the `return` of this hook isn't reported on current master:

https://github.com/gravitational/teleport/blob/12463985c1a743312a195a4704f729c75faa325c/web/packages/teleterm/src/ui/hooks/useLogger.ts#L23-L30

I think I'll let people know on #dev-frontend that they might see more warnings. I thought it'd happen after I enabled `react-hook/rules-of-hooks`, but for some reason it didn't work the way I expected.